control_panel_api/tests/test_views.py

```diff
@@ -123,6 +127,23 @@ def test_update(self):
         self.assertEqual(data['username'], response.data['username'])
         self.assertEqual(data['auth0_id'], response.data['auth0_id'])
 
+    @patch('control_panel_api.models.User.aws_delete_role')
+    @patch('control_panel_api.models.User.helm_create')
+    @patch('control_panel_api.models.User.aws_create_role')
+    def test_aws_error_and_transaction(self, mock_aws_create_role,
+                                       mock_helm_create, mock_aws_delete_role):
+        mock_helm_create.side_effect = CalledProcessError(1, 'foo error')
+
+        data = {'auth0_id': 'github|3', 'username': 'foo'}
+        response = self.client.post(reverse('user-list'), data)
+        self.assertEqual(HTTP_500_INTERNAL_SERVER_ERROR, response.status_code)
+
+        mock_aws_create_role.assert_called()
+        mock_aws_delete_role.assert_called()
```

Comment:
As said in other comment, I wonder if delete the user IAM role on helm error may cause more problems than other.